internal/awslambda/extension/manager_test.go

```diff
@@ -0,0 +1,343 @@
+		// Enforcing that the correct method is used
+		require.Equal(tb, http.MethodPost, r.Method)
+		// Enforcing that the headers exist
+		require.Contains(tb, r.Header, api.LambdaExtensionIdentifierHeaderKey)
+		require.Contains(tb, r.Header, api.LambdaErrorHeaderKey)
+
+		var payload api.ErrorRequest
+		assert.NoError(tb, json.NewDecoder(r.Body).Decode(&payload))
+
+		switch statusCode {
+		case http.StatusAccepted:
+			rw.WriteHeader(statusCode)
+			assert.NoError(tb, json.NewEncoder(rw).Encode(&api.StatusResponse{
+				Status: "accepted",
+			}))
+		case http.StatusBadRequest, http.StatusForbidden:
+			rw.WriteHeader(statusCode)
+			assert.NoError(tb, json.NewEncoder(rw).Encode(&api.ErrorResponse{
+				Message: "Unable to process exit errored shutdown",
+				Type:    "mock-test.error",
+			}))
+		default:
+			rw.WriteHeader(statusCode)
+		}
+	})
+}
+
+func EventNextHandler(tb testing.TB, ctx context.Context, statusCode, shutdownAfter int, delay time.Duration) http.Handler {
+	count := 0
+	return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		// Enforcing the correct method is used
+		require.Equal(tb, http.MethodGet, r.Method)
+		// Enforcing the header exist
+		require.Contains(tb, r.Header, api.LambdaExtensionIdentifierHeaderKey)
+
+		if delay > 1 {
+			<-clock.After(ctx, delay)
+		}
+
+		switch statusCode {
+		case http.StatusOK:
+			payload := &api.EventNextPayload{
+				EventType:          api.Invoke,
+				Deadline:           clock.FromContext(ctx).Now().UTC().Add(100 * time.Millisecond).UnixMilli(),
+				RequestID:          fmt.Sprint(count),
+				InvokedFunctionARN: "local:dev:gostatsd-extention-manager",
+			}
+			if count > shutdownAfter {
+				payload.EventType = api.Shutdown
+				payload.ShutdownReason = "Finished running test cases"
+			}
+			assert.NoError(tb, json.NewEncoder(rw).Encode(payload))
+			rw.Header().Set(api.LambdaExtensionEventIdentifer, fmt.Sprint(count))
+		case http.StatusForbidden:
+			rw.WriteHeader(statusCode)
+			assert.NoError(tb, json.NewEncoder(rw).Encode(&api.ErrorResponse{
+				Message: "Failed to process next handler",
+				Type:    "mock-test.error",
+			}))
+		default:
+			rw.WriteHeader(statusCode)
+		}
+
+		count++
+	})
+}
+
+func TestManagerRegister(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		Scenario   string
+		StatusCode int
+		Expect     error
+	}{
+		{
+			Scenario:   "Registered to Lambda",
+			StatusCode: http.StatusOK,
+			Expect:     nil,
+		},
+		{
+			Scenario:   "Invalid register payload sent",
+			StatusCode: http.StatusBadRequest,
+			Expect:     ErrFailedRegistration,
+		},
+		{
+			Scenario:   "Forbidden request sent to Lambda",
+			StatusCode: http.StatusForbidden,
+			Expect:     ErrFailedRegistration,
+		},
+		{
+			Scenario:   "Unknown error within lambda",
+			StatusCode: http.StatusInternalServerError,
+			Expect:     ErrFailedRegistration,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Scenario, func(t *testing.T) {
```

Comment:
`Parallel()` doesn't inherit - you need to re-call it on the inner tests.

Reply:
You also need to make sure you copy the range value as `tc := tc` otherwise you'll end up with flaky test cases.